llvm/include/llvm/CodeGen/TargetLowering.h

```diff
@@ -549,7 +549,9 @@ class TargetLoweringBase {
   /// several shifts, adds, and multiplies for this target.
   /// The definition of "cheaper" may depend on whether we're optimizing
   /// for speed or for size.
-  virtual bool isIntDivCheap(EVT VT, AttributeList Attr) const { return false; }
+  virtual bool isIntDivCheap(EVT VT, bool IsSigned, AttributeList Attr) const {
```

Comment:
Maybe just pass through the opcode directly than IsSigned (although then you need to decide between an Instruction opcode or ISD opcode)